src/core/public/utils/use_observable_value.ts

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { useMemo, useRef } from 'react';
+import { Observable } from 'rxjs';
+import { map } from 'rxjs/operators';
+import useObservable from 'react-use/lib/useObservable';
+
+/**
+ * A custom hook for subscribing to an Observable that is a function.
+ *
+ * The `useObservable` hook from the `react-use` library doesn't support subscribing to an Observable
+ * that is a function. This hook addresses that limitation by converting the function Observable
+ * into a plain object Observable and then subscribing to its values using `useObservable` from `react-use`.
```

Comment:
```suggestion
 * A custom hook for subscribing to an Observable which value is a function.
 *
 * The `useObservable` hook from the `react-use` library doesn't support subscribing to an Observable
 * which value is a function. This hook addresses that limitation by mapping the Observable value
 * into a plain object so that `useObservable` can correctly emit the value
```